src/ds/helpers.h

```diff
@@ -156,4 +158,69 @@ namespace snmalloc
   {
     UNUSED(t);
   }
+
+  /**
+   * Sometimes we need atomics with trivial initializer.  Unfortunately, this
+   * became harder to accomplish in C++20.  Fortunately, our rules for accessing
+   * these are at least as strong as those required by C++20's atomic_ref:
+   *
+   *   * The objects outlive any references to them
+   *
+   *   * We always access the objects through references (though we'd be allowed
+   *     to access them without if we knew there weren't other references)
+   *
+   *   * We don't access sub-objects at all, much less concurrently through
+   *     other references.
```

Comment:
Is this property true of the recursive version used on platforms without lazy commit?

Reply:
AIUI, yes: in the recursive case, the objects in question are either a pointer to another page or a `T` on a page; we never, say, claim atomic access to the entire array on either variety of page.